lib/tail.js

```diff
@@ -3,64 +3,116 @@
 'use strict';
 
 const events = require('events');
-const childProcess = require('child_process');
-const tailStream = require('fs-tail-stream');
+const fs = require('fs');
+const TailLib = require('tail').Tail;
+const readLastLines = require('read-last-lines');
 const util = require('util');
 const CBuffer = require('CBuffer');
 const byline = require('byline');
-const commandExistsSync = require('command-exists').sync;
+
+const NB_OF_LINE_TO_PREFETCH = 50;
+const TAIL_RETRY_DELAY = 2000;
+const NEW_LINE_REGEX = /[\r]{0,1}\n/;
+const IS_WIN = process.platform === 'win32';
+
+function getPaddedLinePrefix(filePath, filePaths) {
+  const paddingLength = Math.max(...filePaths.map((fileName) => fileName.length));
+  return filePath.padStart(paddingLength, ' ');
+}
+
+const fileLogger = (fileName) => ({
+  info: (...data) => {
+    console.info(fileName, ':', ...data);
+  },
+  error: (...data) => {
+    console.error(fileName, ':', ...data);
+  }
+});
+
+function readLastLinesIfPossible(path, onLineCb) {
+  const logger = fileLogger(path);
+  fs.access(path, fs.constants.R_OK, (checkFileErr) => {
+    if (!checkFileErr) {
+      readLastLines.read(path, NB_OF_LINE_TO_PREFETCH).then((lines) => {
+        lines.split(NEW_LINE_REGEX).forEach(onLineCb);
+      }, (readErr) => logger.error('Failed to prefetch the file content:', readErr));
+    }
+  });
+}
+
+async function tailFile(path, onLineCb) {
+  let tail;
+  const logger = fileLogger(path);
+
+  function rescheduleTail() {
+    // Close the current tail
+    if (tail) {
+      try {
+        tail.unwatch();
+      } catch (ex) {
+        // Failed to shutdown the previous tail, ignore this since we try to stop it
+      }
+    }
+    setTimeout(() => tailFile(path, onLineCb), TAIL_RETRY_DELAY);
+  }
+
+  // Test that the file exists
+  try {
+    await fs.promises.access(path, fs.constants.R_OK);
+  } catch (ex) {
+    logger.error(`can't access to the file: ${ex}`);
+    rescheduleTail();
+    return;
+  }
+
+  try {
+    logger.info('starting to watch the file');
+    tail = new TailLib(path, {
+      useWatchFile: IS_WIN // Use watchfile on windows as a workaround to this issue: https://docs.microsoft.com/en-us/archive/blogs/asiasupp/file-date-modified-property-are-not-updating-while-modifying-a-file-without-closing-it
+    });
+  } catch (ex) {
+    logger.error(`can't watch the file - ${ex}`);
+    rescheduleTail();
+    return;
+  }
+
+  tail.on('line', onLineCb);
+  tail.on('error', (err) => {
+    logger.error('tail failure -', err);
+    rescheduleTail();
+  });
+}
 
 function Tail(path, opts) {
   events.EventEmitter.call(this);
+  const pathArray = Array.isArray(path) ? path : [path]; // Normalize the parameter
+
+  const pushLineToBuffer = (fileName) =>
+    (line) => {
+      if (line === '') {
```

Comment:
I skip the empty lines, because it may introduce some issues when the logger framework adds a char every time `\n` at the end of the file.
More tests need to be done on this because it's not very clean :s

Reply:
It's fixed, it was due to the initial file loading